lib/perl/Genome/Model/Tools/DetectVariants2/Dispatcher.pm

```diff
@@ -300,18 +303,25 @@ sub _dump_dv_cmd {
     return 1;
 }
 
-sub get_relative_path_to_output_directory {
+sub get_relative_path_to_result_directory {
     my $self = shift;
-    my $full_path = shift;
-    my $relative_path = $full_path;
-    my $temp_path = $self->_temp_staging_directory;
-    $relative_path =~ s/$temp_path\/?//;
-    return $relative_path;
+    my $result_path = shift;
+
+    my $relative_path = $result_path;
+
+    my $dispatcher_path = $self->output_directory;
+    if ($relative_path =~ /$dispatcher_path\/?/ ) {
+        $relative_path =~ s/$dispatcher_path\/?//;
+        return $relative_path;
+    }
+
+    return;
```

Comment:
You can just put the ```s///``` inside the if-condition and remove the match.  The substitution will return true if the match portion matched anything.

I know it was there before, but the substitution seems suspect... It's looking for a (partial?) path followed by a slash followed by any character, then cutting it out.  If there was any more of the path after the part that matched, it's now bogus because it's missing that one extra character.